app/src/main/java/com/conupods/MapsActivity.java

```diff
@@ -39,9 +50,32 @@ protected void onCreate(Bundle savedInstanceState) {
     public void onMapReady(GoogleMap googleMap) {
         mMap = googleMap;
 
-        // Add a marker in Sydney and move the camera
-        LatLng sydney = new LatLng(-34, 151);
-        mMap.addMarker(new MarkerOptions().position(sydney).title("Marker in Sydney"));
-        mMap.moveCamera(CameraUpdateFactory.newLatLng(sydney));
+        // Add a marker in starting location and move the camera
+        mMap.addMarker(new MarkerOptions().position(current).title("Marker in Campus"));
+        mMap.moveCamera(CameraUpdateFactory.newLatLng(current));
+    }
+
+    @Override
+    public void onClick(View v) {
+        // handling onClick Events
+        switch (v.getId()) {
```

Comment:
Related to my comment above, having to do a switch for each one. It would be better to use Lambda functions instead above

Reply:
Refactored this whole part to avoid the need for the switch.